### DISCUSSION TITLE
GPL-V3 vs. LGPL-V3

### DISCUSSION THREAD

**Discussion by ghost** [dbus-cxx/dbus-cxx#89] [General]
Why is the Dbus-cxx library licensed under GPL-V3 instead of LGPL-V3?

I am currently working on a commercial product for Linux. But the use of the GPL-V3 in the Dbus-cxx library makes it impossible to distribute the product to customers, without also giving them the source code to the product, which we are not allowed to do.

Is it possible to change the licensing to LGPL-V3?

**Comment by rm5248** [comment]
The library has always been GPL-V3.  I have not yet investigated what(if anything) needs to be done to either re-license or dual-license the software, but I will make some inquiries now.

**Reply by ghost** [reply]
That sounds great :-).

Is there anything I can do to help you in the process?

**Reply by rm5248** [reply]
After doing some research, it does not appear as though I can re-license without the approval of all who have contributed(unless the changes are trivial).  Unfortunately I don't have any up-to-date contact information for Rick Vinyard, but his username appears to be @rvinyard  - perhaps he will see this and chime in?

**Reply by ghost** [reply]
As I can see it, he hasn't contributed with any code since 2010 and he have certainly not in the latest 2.0 version. And if the 2.0 version is a complete implementation, shouldn’t be you that could decide if the license could be changed?

The fact is, if you have changed the name of the repository when you made the 2.0 version, it will have been a whole new project and the change of the license wouldn't have been a problem :-).

**Reply by rm5248** [reply]
The issue is that since it is derived from the older version, I'm not sure what makes it change "enough" for re-licensing to be OK without contacting all of the people who have committed before([OpenSSL had to do this](https://www.openlogic.com/blog/openssl-license-change-cautionary-tale) which lead to some people not liking it). Ultimately this is just a big legal minefield; since Rick has chimed in and said that he is OK with it, I'm comfortable changing the license at this point.

I'll go ahead and make those changes here shortly.  Probably dual-license with LGPL and BSD.

**Reply by ghost** [reply]
Many thanks for the helping in changing the license :-)

**Comment by rvinyard** [comment]
If you want to reticence as LGPL or even BSD I'm fine with that.

**Comment by rvinyard** [comment]
No problem. Sorry about the autocorrect... reticence --> relicense.
